src/main/java/ch/njol/skript/events/EvtPlayerArmorChange.java

```diff
@@ -0,0 +1,90 @@
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.skript.registrations.Classes;
+import com.destroystokyo.paper.event.player.PlayerArmorChangeEvent;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.EquipmentSlot;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.concurrent.CompletionException;
+
+public class EvtPlayerArmorChange extends SkriptEvent {
+
+	private static boolean BODY_SLOT_EXISTS;
+
+	static {
+		if (Skript.classExists("com.destroystokyo.paper.event.player.PlayerArmorChangeEvent")) {
+			Skript.registerEvent("Armor Change", EvtPlayerArmorChange.class, PlayerArmorChangeEvent.class,
+				"[player] armo[u]r change[d] [(from|at) %-equipmentslots%]")
+				.description("Called when armor pieces of a player are changed.")
+				.requiredPlugins("Paper")
+				.keywords("armor", "armour")
+				.examples(
+					"on armor change:",
+						"\tbroadcast the old armor item",
+					"on player armour changed from helmet slot:",
+						"\tbroadcast the new armor item"
+				)
+				.since("2.5, INSERT VERSION (equipment slots)");
+
+			EquipmentSlot bodySlot = null;
+			try {
+				bodySlot = EquipmentSlot.valueOf("BODY");
+			} catch (IllegalArgumentException | NoSuchFieldError | CompletionException ignored) {};
+			BODY_SLOT_EXISTS = bodySlot != null;
+		}
+	}
+
+	private @Nullable Literal<EquipmentSlot> slotLiteral;
+	private @Nullable EquipmentSlot[] slots = null;
+
+	@Override
+	public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResult) {
+		if (args[0] != null) {
+			//noinspection unchecked
+			slotLiteral = (Literal<EquipmentSlot>) args[0];
+			slots = slotLiteral.getArray();
+			for (EquipmentSlot slot : slots) {
+				if (slot == EquipmentSlot.HAND || slot == EquipmentSlot.OFF_HAND || (BODY_SLOT_EXISTS && slot == EquipmentSlot.BODY)) {
+					Skript.error("You can't detect an armor change event for a '" + Classes.toString(slot) + "'.");
+					return false;
+				}
+			}
+		}
+		return true;
+	}
+
+	@Override
+	public boolean check(Event event) {
+		if (!(event instanceof PlayerArmorChangeEvent changeEvent))
```

Comment:
I don't like execution checking events, because it's a wasted check typically. It'll almost never not be the correct event.

Reply:
I don't believe it's actually necessary here